DAOS-1386 test container snapshot info DAOS-1402 DAOS-1390

dinghwah opened the pull request:
Updated file: src/utils/py/daos_api.py
New     file: src/tests/ftest/container/snapshot_info_test.py
New     file: src/tests/ftest/container/snapshot_info_test.yaml

Signed-off-by: Ding Ho <ding-hwa.ho@intel.com>

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,599 @@
+import json
+from apricot import TestWithServers
+from conversion import c_uuid_to_str
+from daos_api import (DaosContext, DaosPool, DaosContainer, DaosSnapshot,
```
(pylint-unused-import) Unused DaosLog imported from daos_api
(pylint-unused-import) Unused DaosContext imported from daos_api

daosbuild1 replied:
(pylint-unused-import) Unused DaosLog imported from daos_api
(pylint-unused-import) Unused DaosContext imported from daos_api

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,599 @@
+        try:
+            for ind in range(0, len(self.container_epoch_list)):
+                epoch = self.container_epoch_list[ind]
+                ss = self.snapshot_list[ind]
```
(pylint-invalid-name) Invalid variable name "ss"

dinghwah replied:
Updated.

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,599 @@
+import os
+import time
+import traceback
+import sys
```
(pylint-unused-import) Unused import sys

daosbuild1 replied:
(pylint-unused-import) Unused import sys

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,599 @@
+import sys
+import random
+import string
+import json
```
(pylint-unused-import) Unused import json

dinghwah replied:
Updated.

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,599 @@
+                .format(str(error)))
+
+        #(6)Destroy the snapshot
+        self.log.info("==(6)Destroy snapshot  epoch: %s",epoch)
```
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 replied:
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,599 @@
+            self.snapshot.destroy(container.coh, epoch)
+            self.log.info("==Snapshot successfully destroyed")
+        except Exception as error:
+                self.fail(
```
(pylint-bad-indentation) Bad indentation. Found 16 spaces, expected 12

daosbuild1 replied:
(pylint-bad-indentation) Bad indentation. Found 16 spaces, expected 12

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,599 @@
+        try:
+            obj.open()
+            snap_handle3 = self.snapshot.open(container.coh,
+                self.snapshot.epoch)
```
(pylint-bad-continuation) Wrong continued indentation (add 30 spaces).

daosbuild1 replied:
(pylint-bad-continuation) Wrong continued indentation (add 30 spaces).

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,599 @@
+            snap_handle3 = self.snapshot.open(container.coh,
+                self.snapshot.epoch)
+            thedata3 = container.read_an_obj(datasize, dkey, akey,
+                obj, snap_handle3.value)
```
(pylint-bad-continuation) Wrong continued indentation (add 29 spaces).

daosbuild1 replied:
(pylint-bad-continuation) Wrong continued indentation (add 29 spaces).

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,599 @@
+            thedata3 = container.read_an_obj(datasize, dkey, akey,
+                obj, snap_handle3.value)
+            self.log.info("-->thedata_after_snapshot.destroied.value= %s"
+                ,thedata3.value)
```
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 replied:
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,599 @@
+            self.log.info("-->thedata_after_snapshot.destroied.value= %s"
+                ,thedata3.value)
+            self.log.info("==>snapshot_epoch=     %s"
+                ,self.snapshot.epoch)
```
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 replied:
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,599 @@
+            self.log.info("==>snapshot_epoch=     %s"
+                ,self.snapshot.epoch)
+            self.log.info("-->snapshot.list(container.coh,epoch)=%s"
+                ,self.snapshot.list(container.coh,epoch))
```
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 replied:
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,599 @@
+    def display_daosContainer(self):
+        self.log.info("==display_daosContainer========================")
+        self.log.info("self.container OBJ=                %s"
+            ,self.container)
```
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 replied:
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,599 @@
+        self.log.info("self.container OBJ=                %s"
+            ,self.container)
+        self.log.info("self.container.context obj=        %s"
+            ,self.container.context)
```
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 replied:
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,599 @@
+        self.log.info("self.container.context obj=        %s"
+            ,self.container.context)
+        self.log.info("self.container.context.libdaos=    %s"
+            ,self.container.context.libdaos)
```
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 replied:
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,599 @@
+        self.log.info("self.container.context.libdaos=    %s"
+            ,self.container.context.libdaos)
+        self.log.info("self.container.context.libtest=    %s"
+            ,self.container.context.libtest)
```
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 replied:
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,599 @@
+        self.log.info("self.container.context.libtest=    %s"
+            ,self.container.context.libtest)
+        self.log.info("self.container.context.ftable=     %s"
+            ,self.container.context.ftable)
```
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 replied:
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,599 @@
+        self.log.info("self.container.context.ftable=     %s"
+            ,self.container.context.ftable)
+        self.log.info("self.container.uuid obj=           %s"
+            ,self.container.uuid)
```
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 replied:
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,599 @@
+        self.log.info("self.container.uuid obj=           %s"
+            ,self.container.uuid)
+        self.log.info("self.container.coh=                %s"
+            ,self.container.coh)
```
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 replied:
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,599 @@
+        self.log.info("self.container.coh=                %s"
+            ,self.container.coh)
+        self.log.info("self.container.poh=                %s"
+            ,self.container.poh)
```
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 replied:
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,599 @@
+        self.log.info("self.container.poh=                %s"
+            ,self.container.poh)
+        self.log.info("self.container.info obj=           %s"
+            ,self.container.info)
```
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 replied:
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,599 @@
+        self.log.info("self.container.info obj=           %s"
+            ,self.container.info)
+        self.log.info("self.container.info.ci_uuid=       %s"
+            ,self.container.info.ci_uuid)
```
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 replied:
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,599 @@
+        self.log.info("self.container.info.ci_uuid=       %s"
+            ,self.container.info.ci_uuid)
+        self.log.info("self.container.info.es_hce=        %s"
+            ,self.container.info.es_hce)
```
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 replied:
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,599 @@
+        self.log.info("self.container.info.es_hce=        %s"
+            ,self.container.info.es_hce)
+        self.log.info("self.container.info.es_lre=        %s"
+            ,self.container.info.es_lre)
```
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 replied:
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,599 @@
+        self.log.info("self.container.info.es_lre=        %s"
+            ,self.container.info.es_lre)
+        self.log.info("self.container.info.es_lhe=        %s"
+            ,self.container.info.es_lhe)
```
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 replied:
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,599 @@
+        self.log.info("self.container.info.es_lhe=        %s"
+            ,self.container.info.es_lhe)
+        self.log.info("self.container.info.es_ghce=       %s"
+            ,self.container.info.es_ghce)
```
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 replied:
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,599 @@
+        self.log.info("self.container.info.es_ghce=       %s"
+            ,self.container.info.es_ghce)
+        self.log.info("self.container.info.es_glre=       %s"
+            ,self.container.info.es_glre)
```
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 replied:
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,599 @@
+            # cleanup the pool
+            pool.disconnect()
+            pool.destroy(1)
+
```
(pylint-trailing-newlines) Trailing newlines

dinghwah replied:
Updated.

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,599 @@
+        self.log.info("self.container.info.es_glre=       %s"
+            ,self.container.info.es_glre)
+        self.log.info("self.container.info.es_ghpce=      %s"
+            ,self.container.info.es_ghpce)
```
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 replied:
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,599 @@
+        self.log.info("self.container.info.es_ghpce=      %s"
+            ,self.container.info.es_ghpce)
+        self.log.info("self.container.info.ci_nsnapshots= %s"
+            ,self.container.info.ci_nsnapshots)
```
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

dinghwah replied:
Updated.

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,599 @@
+        self.log.info("self.container.info.ci_nsnapshots= %s"
+            ,self.container.info.ci_nsnapshots)
+        self.log.info("self.container.info.ci_snapshots=  %s"
+            ,self.container.info.ci_snapshots)
```
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 replied:
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,599 @@
+        self.log.info("self.container.info.ci_snapshots=  %s"
+            ,self.container.info.ci_snapshots)
+        self.log.info("self.container.info.ci_min_slipped_epoch= %s"
+            ,self.container.info.ci_min_slipped_epoch)
```
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

dinghwah replied:
Updated.

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,599 @@
+        try:
+            for ind in range(0, len(self.container_epoch_list)):
+                epoch = self.container_epoch_list[ind]
+                ss = self.snapshot_list[ind]
```
(pylint-invalid-name) Invalid variable name "ss"

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,599 @@
+import sys
+import random
+import string
+import json
```
(pylint-unused-import) Unused import json

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,599 @@
+            # cleanup the pool
+            pool.disconnect()
+            pool.destroy(1)
+
```
(pylint-trailing-newlines) Trailing newlines

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,599 @@
+        self.log.info("self.container.info.es_ghpce=      %s"
+            ,self.container.info.es_ghpce)
+        self.log.info("self.container.info.ci_nsnapshots= %s"
+            ,self.container.info.ci_nsnapshots)
```
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,599 @@
+        self.log.info("self.container.info.ci_snapshots=  %s"
+            ,self.container.info.ci_snapshots)
+        self.log.info("self.container.info.ci_min_slipped_epoch= %s"
+            ,self.container.info.ci_min_slipped_epoch)
```
(pylint-bad-continuation) Wrong continued indentation (add 10 spaces).
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-573/1/execution/node/44/log

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,589 @@
+        try:
+            self.log.info("==(3)Snapshot with an invalid container handler.")
+            coh = container
+            self.snapshot = DaosSnapshot(self.context)
```
(pylint-attribute-defined-outside-init) Attribute 'snapshot' defined outside __init__

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,589 @@
+            coh = container
+            self.snapshot = DaosSnapshot(self.context)
+            self.snapshot.create(coh, epoch)
+        except:
```
(pylint-bare-except) No exception type(s) specified

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,589 @@
+                    thedata, datasize, dkey, akey, obj_cls=obj_cls)
+                obj.close()
+                #Take a snapshot of the container
+                self.snapshot = DaosSnapshot(self.context)
```
(pylint-attribute-defined-outside-init) Attribute 'snapshot' defined outside __init__

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,589 @@
+        self.display_snapshot(snapshot)    #display ss info after ss created
+        return snapshot
+
+    def test_snapshot_create_negtest(self):
```
(pylint-too-many-locals) Too many local variables (25/20)
(pylint-too-many-branches) Too many branches (14/12)
(pylint-too-many-statements) Too many statements (98/50)

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,589 @@
+        try:
+            self.log.info("==(4)Snapshot with a NULL container handler.")
+            coh = None
+            self.snapshot = DaosSnapshot(self.context)
```
(pylint-attribute-defined-outside-init) Attribute 'snapshot' defined outside __init__

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,589 @@
+            coh = None
+            self.snapshot = DaosSnapshot(self.context)
+            self.snapshot.create(coh, epoch)
+        except:
```
(pylint-bare-except) No exception type(s) specified

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,589 @@
+import os
+import time
+import traceback
+import sys
```
(pylint-unused-import) Unused import sys

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,589 @@
+            self.snapshot.destroy(container.coh, epoch)
+            self.log.info("==Snapshot successfully destroyed")
+        except Exception as error:
+                self.fail(
```
(pylint-bad-indentation) Bad indentation. Found 16 spaces, expected 12

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,589 @@
+import sys
+import random
+import string
+import json
```
(pylint-unused-import) Unused import json

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,589 @@
+import json
+from apricot import TestWithServers
+from conversion import c_uuid_to_str
+from daos_api import (DaosContext, DaosPool, DaosContainer, DaosSnapshot,
```
(pylint-unused-import) Unused DaosLog imported from daos_api
(pylint-unused-import) Unused DaosContext imported from daos_api

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,589 @@
+        try:
+            self.log.info("==(5)Snapshot with a NULL epoch = ")
+            invalid_epoch = None
+            self.snapshot = DaosSnapshot(self.context)
```
(pylint-attribute-defined-outside-init) Attribute 'snapshot' defined outside __init__

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,589 @@
+            invalid_epoch = None
+            self.snapshot = DaosSnapshot(self.context)
+            self.snapshot.create(container.coh, invalid_epoch)
+        except:
```
(pylint-bare-except) No exception type(s) specified

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,589 @@
+            snap_handle3 = self.snapshot.open(container.coh,
+                                              self.snapshot.epoch)
+            thedata3 = container.read_an_obj(datasize, dkey, akey,
+                obj, snap_handle3.value)
```
(pylint-bad-continuation) Wrong continued indentation (add 29 spaces).

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,589 @@
+                rand_str = lambda n: ''.join([random.choice(string.lowercase)
+                                              for i in xrange(n)])
+                self.log.info("==(2)Committing %d additional transactions to "
+                              "the same KV.",num_of_write_obj)
```
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,589 @@
+    :avocado: recursive
+    """
+
+    def display_daosContainer(self):
```
(pylint-invalid-name) Invalid method name "display_daosContainer"

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,589 @@
+            pool.destroy(1)
+
+
+    def test_snapshot_info(self):
```
(pylint-too-many-locals) Too many local variables (37/20)
(pylint-too-many-branches) Too many branches (17/12)
(pylint-too-many-statements) Too many statements (128/50)

dinghwah replied:
Updated.

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,589 @@
+            # cleanup the pool
+            pool.disconnect()
+            pool.destroy(1)
+
```
(pylint-trailing-newlines) Trailing newlines

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,589 @@
+                                                obj_cls=obj_cls)
+            obj.close()
+            ##Take a snapshot of the container
+            self.snapshot = DaosSnapshot(self.context)
```
(pylint-attribute-defined-outside-init) Attribute 'snapshot' defined outside __init__

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,589 @@
+            try:
+                reported_epoch = self.snapshot.list(container.coh, epoch)
+                self.log.info("==(4)List snapshot reported_epoch=%s"
+                              ,reported_epoch)
```
(pylint-bad-whitespace) Exactly one space required after comma

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,589 @@
+        try:
+            for ind in range(0, len(container_epoch_list)):
+                epoch = container_epoch_list[ind]
+                ss = snapshot_list[ind]
```
(pylint-invalid-name) Invalid variable name "ss"

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,589 @@
+        """
+        To take a snapshot on the container with current epoch.
+
+        Args: 
```
(style)  trailing whitespace
(pylint-trailing-whitespace) Trailing whitespace

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,589 @@
+
+        Args: 
+            epoch: the container epochfor the snapshot
+        Return: 
```
(style)  trailing whitespace
(pylint-trailing-whitespace) Trailing whitespace

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-573/2/execution/node/44/log

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,547 @@
+        createsetid = self.params.get("setname",
+                                      '/run/poolparams/createset/')
+        createsize = self.params.get("size", '/run/poolparams/createsize/')
+        self.log.info("==In setUp, self.context=", self.context)
```
(pylint-logging-too-many-args) Too many arguments for logging format string

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,547 @@
+            datasize = len(test_data[ind]) + 1
+            try:
+                obj.open()
+                snap_handle = snapshot.open(self.container.coh, current_ss.epoch)
```
(pylint-line-too-long) Line too long (81/80)

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,547 @@
+            self.log.info("==>Negative test 3, expecting failed on taking "
+                          "snapshot with a NULL epoch.")
+        else:
+            self.fail("##(5)Negative test 3 passing, expecting failed on taking "
```
(pylint-line-too-long) Line too long (81/80)

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,547 @@
+            # initialize a python pool object then create the underlying
+            # daos storage
+            self.pool = DaosPool(self.context)
+            self.pool.create(createmode, createuid, creategid, createsize, createsetid, None)
```
(pylint-line-too-long) Line too long (93/80)

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,547 @@
+
+
+
+    def test_snapshot_info(self):
```
(pylint-too-many-locals) Too many local variables (30/20)
(pylint-too-many-branches) Too many branches (14/12)
(pylint-too-many-statements) Too many statements (99/50)

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,547 @@
+        try:
+            snapshot = DaosSnapshot(self.context)
+            snapshot.create(coh, epoch)
+        except:
```
(pylint-bare-except) No exception type(s) specified

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,547 @@
+            self.log.info("==>Negative test 2, expecting failed on taking "
+                          "snapshot on a NULL container.coh.")
+        else:
+            self.fail("##(4)Negative test 2 passing, expecting failed on taking "
```
(pylint-line-too-long) Line too long (81/80)

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,547 @@
+
+
+
+    def test_snapshot_negativecases(self):
```
(pylint-too-many-statements) Too many statements (55/50)

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-573/3/execution/node/44/log

daosbuild1 commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,522 @@
+                      "taking snapshot with a NULL epoch.")
+
+
+    def test_snapshot_info(self):
```
(pylint-too-many-locals) Too many local variables (29/20)
(pylint-too-many-statements) Too many statements (94/50)

saurabhtandan replied:
Seems like this test is doing more than just testing snapshot information functionality. Hence the test name can be a bit misleading. We can change it to something more meaningful here. Like - snapshot_functionality or something. Also same goes for the name of the test and yaml files. The tests in side these files are doing more than just "info" test. So having something more meaning will be nice.
Can be "snapshot_functionality_test.py" If this covers all the important functional tests for snapshot. If you think some better please go ahead with that.

daosbuild1 replied:
(pylint-too-many-locals) Too many local variables (29/20)
(pylint-too-many-statements) Too many statements (94/50)

daosbuild1 replied:
(pylint-too-many-locals) Too many local variables (29/20)
(pylint-too-many-statements) Too many statements (94/50)

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-573/4/execution/node/44/log

saurabhtandan commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,522 @@
+                 DAOS-1395 Test snapshot destroy
+                 DAOS-1402 Test creating multiple snapshots
+        Test Description:
+                (1)Create an object, write random adata into it, and take
```
typo "adata"

dinghwah replied:
Thanks. Upadted.

saurabhtandan commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,522 @@
+                   ==>Repeat step(1) to step(4) for multiple snapshot tests.
+                (5)Verify the snapshots data.
+                (6)Destroy the snapshot.
+                (7)Check if still able to Open the destroied snapshot and
```
typo "destroied"

saurabhtandan commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,522 @@
+                (6)Destroy the snapshot.
+                (7)Check if still able to Open the destroied snapshot and
+                   Verify the snapshot removed from the snapshot list.
+        Use Cases:
```
same as before.

dinghwah replied:
Updated.

saurabhtandan commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,522 @@
+            self.fail(
+                "##Error on snapshot.destroy: %s", str(error))
+
+        #(7)Check if still able to Open the destroied snapshot and
```
typo "destroied"

dinghwah replied:
Updated.

saurabhtandan commented on src/tests/ftest/container/snapshot_info_test.yaml:
```diff
@@ -0,0 +1,32 @@
+# change host names to your reserved nodes, the
+# required quantity is indicated by the placeholders
+hosts:
+    test_machines1:
```
Can be removed. Don't need extra level in the tree for nodes.

saurabhtandan commented on src/tests/ftest/container/snapshot_info_test.yaml:
```diff
@@ -0,0 +1,32 @@
+      test_machines:
+        - boro-A
+        - boro-B
+timeout: 100
```
Is this timeout atleast twice the actual runtime required for each test?

saurabhtandan commented on src/tests/ftest/container/snapshot_info_test.yaml:
```diff
@@ -0,0 +1,32 @@
+        - boro-A
+        - boro-B
+timeout: 100
+faults: !mux
```
Don't need mux here

saurabhtandan commented on src/tests/ftest/container/snapshot_info_test.yaml:
```diff
@@ -0,0 +1,32 @@
+     mode: 511
+   createset:
+     setname: daos_server
+   createsize: !mux
```
don't need mux here

dinghwah commented:
Saurabh, 
Thanks for the review and comments, I have addressed all the comments on the updated test scripts. I also change the test script name to:   snapshot_functionality_test.py and .yaml.
Please review.
Thanks, Ding.

dinghwah commented:
Saurabh, 
Thanks for the review and comments, I have addressed all the comments on the updated test scripts. I also change the test script name to:   snapshot_functionality_test.py and .yaml.
Please review.
Thanks, Ding.

daosbuild1 commented on src/tests/ftest/container/snapshot_functionality_test.py:
```diff
@@ -0,0 +1,529 @@
+          DAOS-1402 Test creating multiple snapshots
+
+    Test Class Description:
+          Start DAOS servers, set up the pool and container for the above 
```
(style)  trailing whitespace
(pylint-trailing-whitespace) Trailing whitespace

daosbuild1 commented on src/tests/ftest/container/snapshot_functionality_test.py:
```diff
@@ -0,0 +1,529 @@
+                      "taking snapshot with a NULL epoch.")
+
+
+    def test_snapshot_info(self):
```
(pylint-too-many-locals) Too many local variables (29/20)
(pylint-too-many-statements) Too many statements (93/50)

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-573/5/execution/node/44/log

daosbuild1 commented on src/tests/ftest/container/snapshot_functionality_test.py:
```diff
@@ -0,0 +1,529 @@
+                      "taking snapshot with a NULL epoch.")
+
+
+    def test_snapshot_info(self):
```
(pylint-too-many-locals) Too many local variables (29/20)
(pylint-too-many-statements) Too many statements (93/50)

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-573/6/execution/node/45/log

dinghwah commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,522 @@
+        set up method
+        """
+        super(Snapshot, self).setUp()
+        # get paths from the build_vars generated by build
```
Thanks for the review and comments,
Please see the new commite (#6) with a new files snapshot_functionality_test.py and .yaml

dinghwah replied:
Updated.

dinghwah commented on src/tests/ftest/container/snapshot_info_test.py:
```diff
@@ -0,0 +1,522 @@
+
+    def display_snapshot(self, snapshot):
+        """
+        To display the snapshot information.
```
Updated.

saurabhtandan commented on src/tests/ftest/container/snapshot_functionality_test.yaml:
```diff
@@ -0,0 +1,31 @@
+    test_machines:
+        - boro-A
+        - boro-B
+timeout: 100
```
Can you please suggest if this timeout is at least twice the total run time of this test?

saurabhtandan commented on src/tests/ftest/container/snapshot_functionality_test.yaml:
```diff
@@ -0,0 +1,31 @@
+        - boro-A
+        - boro-B
+timeout: 100
+faults: !mux
```
Don't think you need mux here. Can you please explain if you have it for any reason?

dinghwah replied:
The run-time is about 250-seconds.  Have updated with 500.

daosbuild1 commented on src/tests/ftest/container/snapshot_functionality_test.py:
```diff
@@ -0,0 +1,532 @@
+                      "taking snapshot with a NULL epoch.")
+
+
+    def test_snapshot_info(self):
```
(pylint-too-many-locals) Too many local variables (29/20)
(pylint-too-many-statements) Too many statements (93/50)

daosbuild1 replied:
(pylint-too-many-locals) Too many local variables (29/20)
(pylint-too-many-statements) Too many statements (93/50)

daosbuild1 replied:
(pylint-too-many-locals) Too many local variables (29/20)
(pylint-too-many-statements) Too many statements (93/50)

daosbuild1 replied:
(pylint-too-many-locals) Too many local variables (29/20)
(pylint-too-many-statements) Too many statements (93/50)

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-573/7/execution/node/44/log

dinghwah commented on src/tests/ftest/container/snapshot_functionality_test.py:
```diff
@@ -0,0 +1,529 @@
+                 DAOS-1395 Test snapshot destroy
+                 DAOS-1402 Test creating multiple snapshots
+        Test Description:
+                (1)Create an object, write random adata into it, and take
```
Thanks.

saurabhtandan commented on src/tests/ftest/server/metadata.yaml:
```diff
@@ -3,10 +3,10 @@
   servers:
-    - boro-A
-    - boro-B
+    - boro-34
```
Please undo the changes and resubmit

saurabhtandan commented on src/tests/ftest/soak/soak.yaml:
```diff
@@ -1,7 +1,7 @@
    test_machines:
         daos_servers:
-            - boro-A
+            - boro-65
```
same as above

daosbuild1 commented on src/include/daos/dtx.h:
```diff
@@ -76,8 +66,8 @@ daos_dti_equal(struct dtx_id *dti0, struct dtx_id *dti1)
-#define DF_DTI		DF_UUID
-#define DP_DTI(dti)	DP_UUID((dti)->dti_uuid)
+#define DF_DTI		DF_UUID"."DF_X64
+#define DP_DTI(dti)	DP_UUID((dti)->dti_uuid), (dti)->dti_hlc
```
(style)  Macros with complex values should be enclosed in parentheses

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-573/8/execution/node/44/log

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-573/9/execution/node/45/log

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-573/10/execution/node/45/log

dinghwah commented on src/tests/ftest/container/snapshot_functionality_test.py:
```diff
@@ -0,0 +1,532 @@
+        finally:
+            # cleanup the pool
+            self.pool.disconnect()
+            self.pool.destroy(1)
```
Thanks for the comment, please see the new commit which has the fixes.

johannlombardi commented on src/tests/ftest/container/snapshot_functionality_test.py:
```diff
@@ -0,0 +1,539 @@
+                   does a commit when the update is complete.
+                (3)Verify the data in the snapshot is the original data.
+                   Get a handle for the snapshot and read the object at dkey,
+                   akey. Compare it to the originally written data.
```
Good. You should just merge this with the previous test and do:
1. write data
2. take snapshot
3. read data back & verify content
4. overwrite data
5. read data from snapshot & verify content
...

dinghwah replied:
Updated.

johannlombardi commented on src/tests/ftest/container/snapshot_functionality_test.py:
```diff
@@ -0,0 +1,539 @@
+        #
+        #Test loop for creat, modify and snapshot object in the DAOS container.
+        #
+        while snapshot_index <= snapshot_loop:
```
sorry, i won't understand what this snapshot loop is about. Could you please elaborate?

johannlombardi commented on src/tests/ftest/container/snapshot_functionality_test.py:
```diff
@@ -0,0 +1,539 @@
+
+            #(2)Make changes to the data object. The write_an_obj function does
+            #   a commit when the update is complete
+            more_transactions = 100
```
sorry, don't get what those transactions are about.

dinghwah replied:
As discussed in the review meeting.

johannlombardi commented on src/tests/ftest/container/snapshot_functionality_test.py:
```diff
@@ -0,0 +1,539 @@
+            #        "epoch list is not the same as the original epoch"
+            #        "snapshotted.")
+            self.log.info("==After 10 additional commits the snapshot is "
+                          "still available")
```
Not sure what this is about.

dinghwah commented on src/tests/ftest/container/snapshot_functionality_test.py:
```diff
@@ -0,0 +1,539 @@
+#!/usr/bin/python
```
Updated.

dinghwah commented on src/tests/ftest/container/snapshot_functionality_test.py:
```diff
@@ -0,0 +1,539 @@
+    """
+    DAOS_PC_RO = int(1 << 0)
+    DAOS_PC_RW = int(1 << 1)
+    DAOS_PC_EX = int(1 << 2)
```
Updated

dinghwah commented on src/tests/ftest/container/snapshot_functionality_test.py:
```diff
@@ -0,0 +1,539 @@
+                self.container.close()
+
+            # wait a few seconds and then destroy
+            time.sleep(5)
```
Removed the unneeded sleep.
Thanks,

daosbuild1 commented on src/tests/ftest/container/snapshot.py:
```diff
@@ -0,0 +1,499 @@
+"""
+from __future__ import print_function
+import os
+import time
```
(pylint-unused-import) Unused import time

daosbuild1 commented on src/tests/ftest/container/snapshot.py:
```diff
@@ -0,0 +1,499 @@
+        data_size = self.params.get("test_datasize",
+                                    '/run/snapshot/*', default=150)
+        snapshot_loop = self.params.get("num_of_snapshot",
+                                    '/run/snapshot/*', default=10)
```
(pylint-bad-continuation) Wrong continued indentation (add 4 spaces).

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-573/14/execution/node/44/log

dinghwah commented on src/tests/ftest/container/snapshot_functionality_test.py:
```diff
@@ -0,0 +1,539 @@
+
+    def take_snapshot(self, container, epoch):
+        """
+        To take a snapshot on the container with current epoch.
```
snapshot took container-handle and container-epoch as input parameters.
and snapshot.epoch is an output parameter.

dinghwah commented on src/tests/ftest/container/snapshot_functionality_test.py:
```diff
@@ -0,0 +1,539 @@
+                          "snapshot with a NULL epoch.")
+        else:
+            self.fail("##(5)Negative test 3 passing, expecting failed on "
+                      "taking snapshot with a NULL epoch.")
```
Snapshot is failing on a NULL epoch (as input parameter).

johannlombardi commented on src/utils/py/daos_api.py:
```diff
@@ -1876,13 +1876,15 @@ def create(self, coh, epoch):
     #  daos_cont_list_snap in src/include/daos_api.h. This must be done for
     #  DAOS-1336 Verify container snapshot info.
-    def list(self, coh):
+    def list(self, coh, epoch=None):
```
From the C API:
  * \param[out]  epochs  preallocated array of epochs to store snapshots.

So this is actually where we return the snapshot list.

johannlombardi commented on src/utils/py/daos_api.py:
```diff
@@ -1876,13 +1876,15 @@ def create(self, coh, epoch):
+            epoch = self.epoch
         num = ctypes.c_uint64(1)
         epoch = ctypes.c_uint64(self.epoch)
         anchor = Anchor()
```
I think we should just report back a dictionary with the list of epochs that have been snapshotted.

johannlombardi commented on src/utils/py/daos_api.py:
```diff
@@ -1893,13 +1895,15 @@ def list(self, coh):
         return epoch.value
 
-    def open(self, coh):
+    def open(self, coh, epoch=None):
```
If the intent is really to open a snapshot, then the epoch should be mandatory there.

johannlombardi commented on src/utils/py/daos_api.py:
```diff
@@ -1908,7 +1912,7 @@ def open(self, coh):
         return txhndl
 
-    def destroy(self, coh, evnt=None):
+    def destroy(self, coh, epoch=None, evnt=None):
```
Same comment here.

johannlombardi commented on src/utils/py/daos_api.py:
```diff
@@ -1917,6 +1921,8 @@ def destroy(self, coh, evnt=None):
         """
         func = self.context.get_function('destroy-snap')
+        if epoch is None:
+            epoch = self.epoch
```
We should remove the notion of self.epoch totally IMO.
